src/authentication.cpp

```diff
@@ -228,23 +227,20 @@ bool irods::s3::authentication::authenticates(
     std::cout << sts << '\n';
     std::cout << "=========================" << std::endl;
 
-    auto irods_user = irods::s3::authentication::get_iRODS_user(&conn, access_key_id).value();
+    auto irods_user = irods::s3::authentication::get_iRODS_user(&conn, access_key_id);
+
+    if (!irods_user) {
+        std::cout << "Authentication Error: No iRODS username mapped to access key ID [" << access_key_id << "]." << std::endl;
+        return std::nullopt;
+    }
+
     auto signing_key = get_user_signing_key(
         irods::s3::authentication::get_user_secret_key(&conn, access_key_id).value(), date, region);
     auto computed_signature = hex_encode(hmac_sha_256(signing_key, sts));
 
-    SwitchUserInput input{};
-    std::strcpy(input.username, irods_user.c_str());
-    std::strcpy(input.zone, conn.clientUser.rodsZone);
-    addKeyVal(&input.options, KW_CLOSE_OPEN_REPLICAS, "");
-
-    if (const auto ec = rc_switch_user(&conn, &input); ec < 0) {
-        std::cout << "Faied to switch users! code [" << ec << "]" << std::endl;
-    }
```

Comment:
I know we just got done ripping out a lot of this kind of stuff in HTTP API, but would it make sense to leave this in as a compiler flag for the good version? :)

Reply:
also.. 'Faied'!

Reply:
I removed those lines because there's no connection pool being used in the project.

I don't think we need to keep that at this time. We can always reference the HTTP API for guidance. I imagine this part of the project is going to need to be refactored even more in the future.